### PR TITLE
Add proxy server options for curl

### PR DESCRIPTION
**Pull request by msiomkin**
Added support for the following curl options:
* CURLOPT_PROXY
* CURLOPT_PROXYPORT
* CURLOPT_PROXYUSERPWD

### CODE REVIEW

**Comment by Totktonada** [comment]
BTW, libcurl aware of environment variables afair. Just note.

See, there are conflicting files.

File: src/httpc.c [hunk 1]
```diff
@@ -320,6 +320,24 @@ httpc_set_ssl_cert(struct httpc_request *req, const char *ssl_cert)
 	curl_easy_setopt(req->curl_request.easy, CURLOPT_SSLCERT, ssl_cert);
 }
 
+void
+httpc_set_proxy(struct httpc_request *req, const char *proxy)
+{
+    curl_easy_setopt(req->curl_request.easy, CURLOPT_PROXY, proxy);
```
Comment (Totktonada) [inline src/httpc.c]: Use tab as the indentation symbol in C sources.

**Comment by Totktonada** [comment]
Also please use httpc: prefix for the commit header.

**Comment by kostja** [comment]
Changes were requested and are not being worked on. Please fix the requested changes and resubmit the pull request.

**Pull request closed without merging**